### ISSUE TITLE
Need resolves with the controller.

### ISSUE THREAD

**Issue by bpmckee** [likeastore/ngDialog#85]
I'm interested in using the 'controller' parameter with ngDialog, however, I need to pass data to the controller.  It would be nice if I could do that using a resolve similar to how ui-router does it.

For example, this could be my controller
```
app.controller('TestCtrl',function($scope, $api, someId){
    $scope.addComment = function(commentText){
        $api.addCommentToSomething(someId, commentText).success(...).error(...);
    };
});
```

I would need to pass in 'someId' somehow, but I may want to do it via resolve like this:
```
ngDialog.open({
    template: 'myTemplate.html',
    controller: 'TestCtrl',
    resolve: {
        someId: ['$myResources', function($myResources){
            return $myResources.getSomeID();
        }
    }
});
```

**Comment by sam-berghmans** [comment]
Is there any update on this? Would be great to have it.

**Comment by rur** [comment]
I need this features so I'll prepare a pull request.

**Comment by gottfrois** [comment]
+1

**Comment by voronianski** [comment]
Feature released in `0.4.0` version.
